tools/depmod.c

```diff
@@ -1479,23 +1482,11 @@ static void depmod_modules_sort(struct depmod *depmod)
 			    depmod->cfg->dirname, order_file, idx);
 			goto corrupted;
 		}
-	}
-	total = idx + 1;
-	idx = 0;
-	fseek(fp, 0, SEEK_SET);
-	while (fgets(line, sizeof(line), fp) != NULL) {
-		size_t len = strlen(line);
-		struct mod *mod;
-
-		idx++;
-		if (len == 0)
-			continue;
 		line[len - 1] = '\0';
-
 		mod = hash_find(depmod->modules_by_uncrelpath, line);
-		if (mod == NULL)
+		if (mod == NULL || mod->sort_idx < 0)
```

Comment:
It's negative if we encounter the line twice, i.e. someone entered the same line multiple times in input file. This could also allow to overflow any kind of counter.

In general, this value is supposed to be negative to be in front of all other indices written so far. So "unsigned int" makes a rather wrong impression on what's going on here.

Reply:
I can strip it out of this comment, but now that you've mentioned it: I didn't notice that it was a bug before as well. It occurred to me because I was trying to think about how to break my new approach.

Reply:
I'd prefer to keep it as it is, because otherwise we would have to keep track which line we have seen so far., bloating code which is removed with the next commit. So ... it's a nice bonus due to rewrite. ;)

Reply:
Oh yes, adding this AFTER my commit makes sense. I thought about adding this check to existing code, which would have been very complicated. Sorry for the noise. Adjusted.